rdsn/src/meta/meta_service.cpp

```diff
@@ -1306,5 +1306,15 @@ void meta_service::on_set_max_replica_count(configuration_set_max_replica_count_
                      server_state::sStateHash);
 }
 
+int32_t meta_service::mutation_2pc_min_replica_count(int32_t app_max_replica_count) const
+{
+    dassert_f(_app_info.max_replica_count > 0);
+if (_opts.mutation_2pc_min_replica_count > 0) {//  >0 means use the user config
+    return _opts.mutation_2pc_min_replica_count;
+} else {// otherwise based on the table max_replica_count
+    return app_max_replica_count <=2 ? 1 :  app_max_replica_count / 2 + 1;
+}
```

Comment:
```suggestion
    if (_opts.mutation_2pc_min_replica_count > 0) { //  > 0 means use the user config
        return _opts.mutation_2pc_min_replica_count;
    } else { // otherwise based on the table max_replica_count
        return app_max_replica_count <= 2 ? 1 :  app_max_replica_count / 2 + 1;
    }
```